src/main/java/io/akka/monitoring/TemperatureMonitoringSetup.java

```diff
@@ -0,0 +1,61 @@
+package io.akka.monitoring;
+
+import akka.Done;
+import akka.javasdk.ServiceSetup;
+import akka.javasdk.annotations.Setup;
+import akka.javasdk.client.ComponentClient;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Sink;
+import io.akka.monitoring.application.IoTDeviceTemperatureStream;
+import io.akka.monitoring.application.TemperatureSummaryAgent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+
+@Setup
+public class TemperatureMonitoringSetup implements ServiceSetup {
+
+  private static final Logger log = LoggerFactory.getLogger(TemperatureMonitoringSetup.class);
+  private final Materializer materializer;
+  private final ComponentClient componentClient;
+  private final IoTDeviceTemperatureStream temperatureStream;
+  private CompletionStage<Done> runningStream;
+
+  public TemperatureMonitoringSetup(Materializer materializer, ComponentClient componentClient) {
+    this.materializer = materializer;
+    this.componentClient = componentClient;
+    this.temperatureStream = new IoTDeviceTemperatureStream(componentClient);
+  }
+
+  @Override
+  public void onStartup() {
+    //run the temperature stream that will simulate IoT device temperature readings
+    this.runningStream = temperatureStream
+      .createStream()
+      .runWith(Sink.ignore(), materializer);
```

Comment:
I think I would just have this as a separate main class that the user needs to run on the side to create traffic. And then expose a POST temperature endpoint where the sensor data is pushed in.

Reply:
It will be clear, but inconvenient, I would prefer to have a simple instruction like: start the service, go to the page, and then wait for auto-updates. I would prefer to make it clearer what is going on in the Readme or the comments.